### ISSUE TITLE
Router can't find child route when navigating from two siblings with default route

### ISSUE THREAD

**Issue by ndcunningham** [angular/angular#9648]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
```

**Current behavior**
Currently, when navigating from a child component to another child component we get an error message saying ` Cannot match any routes: 'routeName'`

**Expected/desired behavior**
Child to navigate to any other sibling component

**Reproduction of the problem**
1. Navigate to Login
2. Click Login
3. Navigate Protected Parent/Child route
4. Click on `Click `to navigate to the secondChild Component

[How it works without a default route](http://plnkr.co/edit/hqMaLQ?p=preview)

[How it works with a default route](http://plnkr.co/edit/FnNsaeS43YkgesfXNGsL) **Bug**

![image](https://cloud.githubusercontent.com/assets/338948/16394595/0b817af0-3c84-11e6-9a14-36f0489e85f2.png)

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.3
 
* **Browser:** Chrome 52
 
* **Language:** TypeScript 1.8.10

**Comment by fknop** [comment]
It looks like it works with `./child` instead of `../child`, which is probably not intended.

**Comment by ndcunningham** [comment]
> One last thing for your plnkr, your route paths and navigation links do not match. You have child and try to navigate to Child.

Thanks @fknop 

DI for?

**Comment by fknop** [comment]
The `Router` and `ActivatedRoute` were missing for me.

**Comment by ndcunningham** [comment]
@fknop  confirmed 
`this.router.navigate(['./child'], { relativeTo: this.route });` works instead of 
`this.router.navigate(['../child'], { relativeTo: this.route });`

**Comment by fknop** [comment]
I looked into the snapshot, the property `_lastPathIndex` for the `ActivatedRoute` for `ParentHome` looks wrong, it's `0`, the other children are `1`. I'm not sure what this property means exactly tho.

**Comment by ndcunningham** [comment]
hmmm...
maybe @brandonroberts can shed some light

**Comment by ndcunningham** [comment]
Additionally, I realized an inconsistency.
http://plnkr.co/edit/FnNsaeS43YkgesfXNGsL?p=preview

I updated my bug plunker to show that only the default route will navigate to a child using
`this.router.navigate(['./child'], { relativeTo: this.route });`

If this is followed by any child routes that have a **path** which is not `''` the _correct_ syntax works that is: `this.router.navigate(['../child'], { relativeTo: this.route });`

Amending my previous instructions on the Second Child shows two types of navigation one while 1 `.` and another with `..`. To reflect my claim earlier.

@fknop I think it's based on where the router is looking currently which is wrong for a default route
> I looked into the snapshot, the property _lastPathIndex for the ActivatedRoute of ParentHome looks wrong, it's 0, the other children are 1

tldr;
**Default** parent routes with no path can only navigate to a child using ``this.router.navigate(['./child'], { relativeTo: this.route });`` While all other routes which have paths `this.router.navigate(['../sibling'], { relativeTo: this.route });` will work

**Comment by vsavkin** [comment]
I tried this plunker with the latest version of the router: http://plnkr.co/edit/FnNsaeS43YkgesfXNGsL?p=preview, and the example works for me

Let me clarify one thing though to make sure we are on the same page. The router navigation is url-based, not route-based.

Say we have this configuration

```
{path: 'parent', children: [
{path: '', component: A},
{path: 'b', component: B},
{path: 'c', component: C}
]}
```

When navigating to '/parent', the router will render A. To navigate to 'parent/b', we would use this link:
`./b`. Now, say we navigated to 'parent/b' and the router rendered B. To navigate to 'parent/c', we would use this link: `../c`.

**Comment by fknop** [comment]
@vsavkin Thanks for the clarification. Indeed, it makes sense now.

**Comment by ndcunningham** [comment]
oh so navigating from no path to an actual path requires `./path` while navigating between two *defined* path components on the same level requires `../path`

Are we on the same page?

**Comment by fknop** [comment]
@ndcunningham It's URL based so `./` add a segment, `../` removes a segment.

`a/b` `->` `./c` `->` `a/b/c` 
`a/b` `->` `../c` `->` `a/c`

**Comment by ndcunningham** [comment]
so yes to my statement then.

thanks @fknop @vsavkin
